src/django_components/extensions/view.py

```diff
@@ -0,0 +1,80 @@
+from typing import TYPE_CHECKING, Any, Protocol, cast
+
+from django.http import HttpRequest, HttpResponse
+from django.views.generic import View
+
+from django_components.extension import BaseExtensionClass, ComponentExtension
+
+if TYPE_CHECKING:
+    from django_components.component import Component
+
+
+class ViewFn(Protocol):
+    def __call__(self, request: HttpRequest, *args: Any, **kwargs: Any) -> Any: ...  # noqa: E704
+
+
+class ComponentView(BaseExtensionClass, View):
+    """
+    Subclass of `django.views.View` where the `Component` instance is available
+    via `self.component`.
+    """
+
+    # NOTE: This attribute must be declared on the class for `View.as_view()` to allow
+    # us to pass `component` kwarg.
+    component = cast("Component", None)
+
+    def __init__(self, component: "Component", **kwargs: Any) -> None:
+        BaseExtensionClass.__init__(self, component)
+        View.__init__(self, **kwargs)
+
+    # NOTE: The methods below are defined to satisfy the `View` class. All supported methods
+    # are defined in `View.http_method_names`.
+    #
+    # Each method actually delegates to the component's method of the same name.
+    # E.g. When `get()` is called, it delegates to `component.get()`.
+
+    def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
```

Comment:
Previously, these methods like `get`, `post`, etc were defined dynamically by using the `View.http_method_names` variable.

But it wasn't very readable and intuitive what's going on, so instead I expanded.